sys/kern/Makefile

```diff
@@ -79,7 +79,8 @@ SOURCES-LOCKDEP = \
 	lockdep.c
 
 SOURCES-KGPROF = \
-	kgprof.c
+	kgprof.c \
+	mcount.c
 
 FORMAT-EXCLUDE = sysent.h
 
```

Comment:
You have missed that `mcount.c` should not be profiled.

```
diff --git a/sys/kern/Makefile b/sys/kern/Makefile
index 9aecd6ae..83f50efe 100644
--- a/sys/kern/Makefile
+++ b/sys/kern/Makefile
@@ -94,3 +94,4 @@ sysent.h $(SYSCALL_H) $(SYSCALLARGS_H): syscalls.conf syscalls.master
        $(TOPDIR)/sys/script/makesyscalls.sh $^
 
 kasan.o : CFLAGS_KASAN =
+mcount.o : CFLAGS_KGPROF =
```

Reply:
The attribute `__no_instrument_kgprof` already guarantees it, we do not have to clear flags for this file

Reply:
Are you sure?
I have pulled latest changes from your branch, here is my diff:
```
diff --git a/config.mk b/config.mk
index c72b5e13..2a2d8f82 100644
--- a/config.mk
+++ b/config.mk
@@ -25,4 +25,4 @@ VERBOSE ?= 0
 CLANG ?= 0
 LOCKDEP ?= 0
 KASAN ?= 0
-KGPROF ?= 0
+KGPROF ?= 1
```

I have launched system with: `./launch --board=malta -d init=/bin/ksh` and I'm waiting....
Here is a backtrace:
```
#0  0xc010be00 in __cyg_profile_func_enter (self=self@entry=0xc010c068 <spin_lock>, from=from@entry=0xc010be2c <__cyg_profile_func_enter+84>) at /home/pjas/mimiker-prof/sys/kern/mcount.c:95
#1  0xc010be2c in spin_lock (s=0xc017e48c <mcount_lock>) at /home/pjas/mimiker-prof/include/sys/spinlock.h:61
#2  __cyg_profile_func_enter (self=0xc010c068 <spin_lock>, from=0xc010be2c <__cyg_profile_func_enter+84>) at /home/pjas/mimiker-prof/sys/kern/mcount.c:105
#3  0x00000000 in ?? ()
```
It looks like deadlock on `mcount_lock`.

Reply:
My bad, you are right